common/lib/api-client/middleware/request-headers.js

```diff
@@ -1,8 +1,12 @@
 const getRequestHeaders = require('../request-headers')
 
 module.exports = {
-  name: 'req-headers',
+  name: 'request-headers',
   req: function req(payload = {}) {
+    if (payload.res) {
```

Comment:
This might be nicer to combine with the following lines:

```javascript
const { req, res } = payload

if (!req || res) {
  return payload
}

...
```

Reply:
https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/801#discussion_r492511072

I think it would be better to remove the unnecessary `req` check ;)

Reply:
It feels cleaner and easier to understand to check for a request if we're setting things on the request, no? 

Rather than assuming that the inclusion of a  response means it is a request.

Reply:
> Rather than assuming that the inclusion of a response means it is a request.
I don't understand this comment

The req object is there is pass the request args to jsonApi.axios. In the case of the cached value being returned, we don't want to invoke  jsonApi.axios.

The response handler looks at payload.res. Therefore the cache middleware return the payload with the res property attached. The other middleware know not to bother doing anything and just return the payload as is if that property is present. The payload.res checks are valid. The payload.req ones are pointless.